sdgym/sdgym_result_explorer/result_explorer.py

```diff
@@ -0,0 +1,88 @@
+"""SDGym Results Explorer for accessing and managing benchmark results."""
+
+import os
+
+from sdgym.benchmark import DEFAULT_DATASETS
+from sdgym.datasets import get_dataset_paths, load_dataset
+from sdgym.s3 import _get_s3_client, is_s3_path
+from sdgym.sdgym_result_explorer.result_handler import LocalResultsHandler, S3ResultsHandler
+
+
+def _validate_path(path, aws_access_key_id=None, aws_secret_access_key=None):
+    """Validates the provided path to ensure it is either a local directory or an S3 bucket."""
+    if is_s3_path(path):
+        s3_client = _get_s3_client(path, aws_access_key_id, aws_secret_access_key)
```

Comment:
I think this was confusing because it is validating but also being used to determine if the path ios local or on s3. Can we just determine if it's an s3 path, and if so then get the client all in the init. This function can just be a validation function for local paths